### DISCUSSION TITLE
Modal pagination

### DISCUSSION THREAD

**Discussion by dreamealist** [twilio-labs/paste#3088] [General] [answered]
Sorry for bombarding the team with too many posts today 😆
I had a quick question regarding modal pagination use cases. Specifically, I was wondering if it's possible to have a modal automatically transition to a second page when the content is too long for a single modal. I'm not sure if this is a built-in feature of the Paste modal. If not, would it be easy for our developers to customize the existing modal component?

Additionally, I noticed that the new modal component has the "back" button when you enable the right-side button. Is it intended to be used for the second page of a modal? Could you please provide any recommendation or clarity on this? Thanks in advance for your help!

<img width="1297" alt="Screenshot 2023-03-10 at 3 29 28 PM" src="https://user-images.githubusercontent.com/13625142/224427433-100c3103-f7da-413f-b230-ac201a6a9160.png">
In this use case, when the user chooses "webhook", the modal transitions to the next page to show the detailed configuration.

**Comment by loreina** [comment, marked as answer]
Hi Jiyeon!

The height of the Modal will grow with the amount of content in it and when it reaches 90% of the height of the viewport, the body will begin to scroll. From your screenshots, you should be able to continue with a single content area. 

And yes, Modal footer actions can include a back button for multiple "pages" of content. As per our [docs site](https://paste.twilio.design/components/modal#when-to-use-modal), a Modal can be used to guide the user through a complex workflow with a series of steps, while still maintaining context of the task that initiated the modal. Hope this helps!
